### PR TITLE
To refactor multi tensor SGD to functional API

### PR DESCRIPTION
**Pull request by iramazanli**
Note that for multi-tensor functional API on SGD, we are providing momentum_buffer_list in the states parameter. That's why we don't have this argument here, however in contrast we have states parameter here which carries this missed information.

This change reflects it in the line for single tensor SGD : 
https://github.com/pytorch/pytorch/blob/0b8a7daa2a7e765270b85234b868b5897a9822bb/torch/optim/sgd.py#L120

that we don't have in multi-tensor API.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60964
* #60963
* **#60962**
* #60961
* #60960

### CODE REVIEW

File: torch/optim/_multi_tensor/sgd.py [hunk 1]
```diff
@@ -105,51 +101,21 @@ def step(self, closure=None):
...
-                else:
-                    grads = bufs
-
-            if not has_sparse_grad:
-                torch._foreach_add_(params_with_grad, grads, alpha=-group['lr'])
-            else:
-                # foreach APIs dont support sparse
-                for i in range(len(params_with_grad)):
-                    params_with_grad[i].add_(grads[i], alpha=-group['lr'])
+            F.sgd(params_with_grad,
```
Comment (albanD) [inline torch/optim/_multi_tensor/sgd.py:110]: What remains of the `optim._multi_tensor.SGD()` class seems to be exactly the same as the `optim.SGD()` one.
Do you think we could refactor this code to have a single implementation (and maybe a new attribute that hold the functional step function?

File: torch/optim/_multi_tensor/_functional.py [hunk 2]
```diff
@@ -299,3 +299,58 @@ def adamw(params: List[Tensor],
 
     step_size = [-1 * (lr / bc) for bc in bias_correction1]
     torch._foreach_addcdiv_(params, exp_avg, denom, step_size)
+
+
+def sgd(params: List[Tensor],
```
Comment (albanD) [inline torch/optim/_multi_tensor/_functional.py:304]: Could you re-order these arguments to match the sgd() API from the non-multi-tensor one?
That should allow us to use them interchangeably which will make switching from one to the other much simpler!

Comment (iramazanli) [reply torch/optim/_multi_tensor/_functional.py:304]: Thanks for the feedback, i did re-ordering as much as possible. But please note that, we have states and has_sparse_grad parameters here which is not existing in single tensor API.

File: torch/optim/_multi_tensor/sgd.py (hunk 1 above)
Comment (iramazanli) [reply torch/optim/_multi_tensor/sgd.py:110]: This is actually very good point, and i would love to have further discussions about this. But currently there are certain differences with multi_tensor and non-multi_tensor optimizers. For example: 

For SGD we have :  https://github.com/pytorch/pytorch/blob/bfe03120eebc8b2b7c256d08887a782de9b6177c/torch/optim/sgd.py#L120

For ASGD: https://github.com/pytorch/pytorch/blob/bfe03120eebc8b2b7c256d08887a782de9b6177c/torch/optim/asgd.py#L88

Which we don't need them in multi-tensor case. We will return back to functional optimizers when discussions of functional Optimizers finalizes in Vmap side. Maybe we can investigate your suggestion in that time ?

Comment (albanD) [reply torch/optim/_multi_tensor/sgd.py:110]: Ok

**Pull request closed without merging**